doc/developers/develop.rst

```diff
@@ -519,7 +519,27 @@ which returns the new values for your estimator's tags. For example::
             return tags
 
 You can create a new subclass of :class:`~sklearn.utils.Tags` if you wish to add new
-tags to the existing set.
+tags to the existing set. Note that all attributes that you add in a child class need
+to have a default value. It can be of the form::
+
+    from dataclasses import dataclass, asdict
+
+    @dataclass
+    class MyTags(Tags):
+        my_tag: bool = True
+
+    class MyEstimator(BaseEstimator):
+        def __sklearn_tags__(self):
+            tags_orig = super().__sklearn_tags__()
+            tags = MyTags(**asdict(tags_orig))
+            tags.input_tags = tags_orig.input_tags
+            tags.target_tags = tags_orig.target_tags
+            tags.classifier_tags = tags_orig.classifier_tags
+            tags.regressor_tags = tags_orig.regressor_tags
+            tags.transformer_tags = tags_orig.transformer_tags
```

Comment:
With `MyTags(**asdict(tags_orig))` do we need to set all these afterwards?

Reply:
Yes, `asdict` goes recursive, which then makes `tags.input_tags` a dictionary instead of a `InputTags` instance.

Reply:
Can we either do:

```python
tags = MyTags(**tags_org.__dict__)
```

or: (Copied from the [docs of `asdict`](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict)) 

```python
tags_org_dict = {field.name: getattr(tags_orig, field.name) for field in fields(tags_orig)}
tags = MyTags(**tags_org_dict)
```

Reply:
They don't have a `__dict__`, but the second one works.